DataManagementSystem/Client/ConsistencyChecks.py

```diff
@@ -0,0 +1,836 @@
+    else:
+      value = 0
+    self._prod = value
+  def get_prod( self ):
+    """ Getter """
+    return self._prod
+  prod = property( get_prod, set_prod )
+
+  def set_fileType( self, value ):
+    """ Setter """
+    fts = [ft.upper() for ft in value]
+    self._fileType = fts
+  def get_fileType( self ):
+    """ Getter """
+    return self._fileType
+  fileType = property( get_fileType, set_fileType )
+
+  def set_fileTypesExcluded( self, value ):
+    """ Setter """
+    fts = [ft.upper() for ft in value]
+    self._fileTypesExcluded = fts
+  def get_fileTypesExcluded( self ):
+    """ Getter """
+    return self._fileTypesExcluded
+  fileTypesExcluded = property( get_fileTypesExcluded, set_fileTypesExcluded )
+
+  def set_lfns( self, value ):
+    """ Setter """
+    if type( value ) == type( "" ):
+      value = [value]
+    value = [v.replace( ' ', '' ).replace( '//', '/' ) for v in value]
+    self._lfns = value
+  def get_lfns( self ):
+    """ Getter """
+    return self._lfns
+  lfns = property( get_lfns, set_lfns )
+  
+  ###############################################################################################
+  #
+  #  This part was backported from DataIntegrityClient
+  #
+  #
+  #  This section contains the specific methods for LFC->SE checks
+  #
+
+  def catalogDirectoryToSE( self, lfnDir ):
+    """ This obtains the replica and metadata information from the catalog for the supplied directory and checks against the storage elements.
+    """
+    gLogger.info( "-" * 40 )
+    gLogger.info( "Performing the LFC->SE check" )
+    gLogger.info( "-" * 40 )
+    if type( lfnDir ) in types.StringTypes:
+      lfnDir = [lfnDir]
+    res = self.__getCatalogDirectoryContents( lfnDir )
+    if not res['OK']:
+      return res
+    replicas = res['Value']['Replicas']
+    catalogMetadata = res['Value']['Metadata']
+    res = self.__checkPhysicalFiles( replicas, catalogMetadata )
+    if not res['OK']:
+      return res
+    resDict = {'CatalogMetadata':catalogMetadata, 'CatalogReplicas':replicas}
+    return S_OK( resDict )
+
+  def catalogFileToSE( self, lfns ):
+    """ This obtains the replica and metadata information from the catalog and checks against the storage elements.
+    """
+    gLogger.info( "-" * 40 )
+    gLogger.info( "Performing the LFC->SE check" )
+    gLogger.info( "-" * 40 )
+    if type( lfns ) in types.StringTypes:
+      lfns = [lfns]
+    res = self.__getCatalogMetadata( lfns )
+    if not res['OK']:
+      return res
+    catalogMetadata = res['Value']
+    res = self.__getCatalogReplicas( catalogMetadata.keys() )
+    if not res['OK']:
+      return res
+    replicas = res['Value']
+    res = self.__checkPhysicalFiles( replicas, catalogMetadata )
+    if not res['OK']:
+      return res
+    resDict = {'CatalogMetadata':catalogMetadata, 'CatalogReplicas':replicas}
+    return S_OK( resDict )
+
+  def checkPhysicalFiles( self, replicas, catalogMetadata, ses = [] ):
```

Comment:
super mega uber dangerous to have [] as default attribute. Use None instead. 

Illustration:
```
In [17]: def myMethod(a = []):
   ....:     a.append(0)
   ....:     return a
   ....: 

In [18]: myMethod()
Out[18]: [0]

In [19]: myMethod()
Out[19]: [0, 0]
```